pkg/intermediate/aggregate.go

```diff
@@ -482,14 +491,35 @@ func (a *AggregationProcess) aggregateRecords(incomingRecord, existingRecord ent
 		return nil
 	}
 	isLatest := false
+	var prevFlowEndSeconds, flowEndSecondsDiff uint32
 	if ieWithValue, _, exist := incomingRecord.GetInfoElementWithValue("flowEndSeconds"); exist {
 		if existingIeWithValue, _, exist2 := existingRecord.GetInfoElementWithValue("flowEndSeconds"); exist2 {
 			incomingVal := ieWithValue.GetUnsigned32Value()
 			existingVal := existingIeWithValue.GetUnsigned32Value()
-			if incomingVal > existingVal {
+			if incomingVal >= existingVal {
 				isLatest = true
 				existingIeWithValue.SetUnsigned32Value(incomingVal)
 			}
+			// Update the flowEndSecondsFromSource/DestinationNode fields, and compute
+			// the time difference between the incoming record and the last record.
+			if fillSrcStats {
+				prevFlowEndSeconds = a.updateFlowEndSecondsFromNodes(incomingRecord, existingRecord, true, incomingVal)
+			}
+			if fillDstStats {
+				prevFlowEndSeconds = a.updateFlowEndSecondsFromNodes(incomingRecord, existingRecord, false, incomingVal)
+			}
+			// For inter-node flows, skip the aggregation process if the incoming record
```

Comment:
Change: Include the consideration of intra-node, we skip the aggregation for intra-node if we receive the same record as the last one. (normally we will receive one record from source node, one record from destination node. Here source node and destination is the same node, so we will receive two identical records from the node.

Reply:
For intra-node, do you mean we will receive two records for each flow?

Reply:
I made it wrong in the comment. Sorry for the confusion. I've corrected it. For both intra-node and inter-node, we will send warning if the flowEndSeconds value is not the latest.